tests/specs/mcd/verification.k

```diff
@@ -23,13 +23,23 @@ module VERIFICATION-SYNTAX
-    rule #WordPackAddrUInt48UInt48 ( A , X , Y ) => Y *Int pow208 +Int X *Int pow160 +Int A requires #rangeAddress(A)  andBool #rangeUInt(48, X) andBool #rangeUInt(48, Y)
+    syntax Int ::= #WordPackUInt48UInt48     (       Int , Int ) [function, no-evaluators, smtlib(WordPackUInt48UInt48)]
+                 | #WordPackAddrUInt48UInt48 ( Int , Int , Int ) [function, no-evaluators, smtlib(WordPackAddrUInt48UInt48)]
+                 | #WordPackAddrUInt8        (       Int , Int ) [function, no-evaluators, smtlib(WordPackAddrUInt8)]
+ // -----------------------------------------------------------------------------------------------------------------
+    // rule #WordPackUInt48UInt48     (            UINT48_1 , UINT48_2 ) => UINT48_2 *Int pow48 +Int UINT48_1                        requires #rangeUInt(48, UINT48_1) andBool #rangeUInt(48, UINT48_2)
+    // rule #WordPackAddrUInt48UInt48 (     ADDR , UINT48_1 , UINT48_2 ) => UINT48_2 *Int pow208 +Int UINT48_1 *Int pow160 +Int ADDR requires #rangeAddress(ADDR) andBool #rangeUInt(48, UINT48_1) andBool #rangeUInt(48, UINT48_2)
+    // rule #WordPackAddrUInt8        (     ADDR , UINT8               ) => UINT8 *Int pow160 +Int ADDR                              requires #rangeAddress(ADDR) andBool #rangeUInt(8, UINT_8)
+
+    rule maxUInt48 &Int  UINT48_UINT48             => UINT48_1 requires UINT48_UINT48 ==Int #WordPackUInt48UInt48(UINT48_1, UINT48_2) andBool #rangeUInt(48, UINT48_1) andBool #rangeUInt(48, UINT48_2) [smt-lemma]
+    rule maxUInt48 &Int (UINT48_UINT48 /Int pow48) => UINT48_2 requires UINT48_UINT48 ==Int #WordPackUInt48UInt48(UINT48_1, UINT48_2) andBool #rangeUInt(48, UINT48_1) andBool #rangeUInt(48, UINT48_2) [smt-lemma]
+
+    rule maxUInt160 &Int  ADDR_UINT48_UINT48              => ADDR     requires ADDR_UINT48_UINT48 ==Int #WordPackAddrUInt48UInt48(ADDR, UINT48_1, UINT48_2) andBool #rangeAddress(ADDR) andBool #rangeUInt(48, UINT48_1) andBool #rangeUInt(48, UINT48_2) [smt-lemma]
+    rule maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow160) => UINT48_1 requires ADDR_UINT48_UINT48 ==Int #WordPackAddrUInt48UInt48(ADDR, UINT48_1, UINT48_2) andBool #rangeAddress(ADDR) andBool #rangeUInt(48, UINT48_1) andBool #rangeUInt(48, UINT48_2) [smt-lemma]
+    rule maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow208) => UINT48_2 requires ADDR_UINT48_UINT48 ==Int #WordPackAddrUInt48UInt48(ADDR, UINT48_1, UINT48_2) andBool #rangeAddress(ADDR) andBool #rangeUInt(48, UINT48_1) andBool #rangeUInt(48, UINT48_2) [smt-lemma]
+
+    rule maxUInt160 &Int  ADDR_UINT8              => ADDR  requires ADDR_UINT8 ==Int #WordPackAddrUInt8(ADDR, UINT8) andBool #rangeAddress(ADDR) andBool #rangeUInt(8, UINT8) [smt-lemma]
+    rule maxUInt8   &Int (ADDR_UINT8 /Int pow160) => UINT8 requires ADDR_UINT8 ==Int #WordPackAddrUInt8(ADDR, UINT8) andBool #rangeAddress(ADDR) andBool #rangeUInt(8, UINT8) [smt-lemma]
```

Comment:
How could this work?  Is it what @denis-bogdanas fixed recently?

Reply:
BTW, why did you write these lemmas in this way?  I mean, what's the problem of writing, e.g.,:
```
rule maxUInt48 &Int  #WordPackUInt48UInt48(UINT48_1, UINT48_2)  => UINT48_1 requires #rangeUInt(48, UINT48_1) andBool #rangeUInt(48, UINT48_2)
```
(no `[smt-lemma]`)?

Reply:
Another question.  Could you explain why the commented lemmas don't work?  I guess that z3 is not capable of reasoning about the bitwise operations using the nonlinear expressions, but want to double-check.

Reply:
No, my fix is unrelated and is not merged. I'm surprised this works, but in case it does, seems correct.

Reply:
A worst case scenario: It could be the case that this smt-lemma is encoded incorrectly, leading to a _false_ statement, which will make all z3 queries always _unsat_, making all specs to be blindly passed.  So, we might be better to double-check how this is exactly handled by the java backend.

Reply:
@daejunpark I cannot write it how you would like (as `maxUInt48 &Int #WordPackUInt48UInt48(UINT48_1, UINT48_2) => ...`), because the actual term looks like `maxUInt48 &Int #lookup(STORAGE, SOME_KEY)`, and only via the side-condition do we see that `#lookup(STORAGE, SOME_KEY)` is equal `SOME_VARIABLE`, and `SOME_VARIABLE` is equal to `#WordPackUInt48UInt48(UINT48_1, UINT48_2)`. So the unification/matching just doesn't seem to apply the rule as needed.

Reply:
Oh I see. Then this makes sense.